diff --git a/CMakeLists.txt b/CMakeLists.txt
index e8adf23d9..598680e19 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,14 @@ PROJECT(embree${EMBREE_VERSION_MAJOR})

 CMAKE_MINIMUM_REQUIRED(VERSION 2.8.11)

+if(WIN32)
+    # Enforce static linking of multithread MSVC runtime environment.
+    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /MT" CACHE INTERNAL "" FORCE)
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /MTd" CACHE INTERNAL "" FORCE)
+    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} /MT" CACHE INTERNAL "" FORCE)
+    set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} /MTd" CACHE INTERNAL "" FORCE)
+endif()
+
 # find git version
 IF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/.git)
   FIND_PACKAGE(Git)
